### ISSUE TITLE
Viewport hogged when embedded in iframe

### ISSUE THREAD

**Issue by 1ec5** [ZeLonewolf/openstreetmap-americana#884]
When OSM Americana is embedded in an iframe, it hogs the viewport by scrolling into view on load. The OSMUS website is having to work around this issue by scrolling back to the top of the page.

The culprit appears to be this code that focuses the `<canvas>` element just after loading the shield rendering plugin:

https://github.com/ZeLonewolf/openstreetmap-americana/blob/09cf915acb8597359894b94ed20eeabbbca525b3/src/americana.js#L94

This code is an outgrowth of #61, which focused the canvas in order to ensure that GL JS’s keyboard shortcuts would work immediately on load. That behavior is reasonable, but it shouldn’t happen if the page is in a frame.

**Comment by ZeLonewolf** [comment]
Can the focus be involved selectively?

**Comment by 1ec5** [comment]
Yes, by checking the window’s [parent](https://developer.mozilla.org/en-US/docs/Web/API/Window/parent):

```js
let isEmbedded = window.parent !== window
```

**Comment by jleedev** [comment]
Honestly the standalone and embedded versions should be different pages, but `self === top` is a simple check.

Could also use the [autofocus] attribute to let the browser handle it.

[autofocus]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus

**Comment by quincylvania** [comment]
> Honestly the standalone and embedded versions should be different pages

Agreed, the embedded map on openstreetmap.us has different requirements. The Samples button isn't needed, for example. But making the main version embeddable is still a nice practice in case anyone has a use case for it.
